File: apps/ios/Voiid/Voiid/Main/AIChatView.swift
//
//  AIChatView.swift
//  Voiid
//
//  In-app AI assistant (Figma Screen-8). ChatGPT-style conversation on dummy responses.
//  User bubbles right (#C8C8C8), AI bubbles left (#FCF4F8), with a "thinking" indicator.
//

import SwiftUI

struct AIChatView: View {
    @EnvironmentObject var ai: AIStore
    @State private var draft = ""

    var body: some View {
        VStack(spacing: 0) {
            header
            ScrollViewReader { proxy in
                ScrollView {
                    LazyVStack(spacing: VoiidSpacing.sm) {
                        ForEach(ai.messages) { m in
                            aiBubble(m).id(m.id)
                        }
                        if ai.thinking { TypingBubble().id("thinking") }
                    }
                    .padding(.horizontal, VoiidSpacing.md)
                    .padding(.vertical, VoiidSpacing.md)
                }
                .onChange(of: ai.messages.count) { _, _ in withAnimation { proxy.scrollTo(ai.messages.last?.id, anchor: .bottom) } }
                .onChange(of: ai.thinking) { _, v in if v { withAnimation { proxy.scrollTo("thinking", anchor: .bottom) } } }
            }
            inputBar
        }
        .background(VoiidColor.background.ignoresSafeArea())
    }

    private var header: some View {
        HStack(spacing: VoiidSpacing.sm) {
            Image(systemName: "sparkles").font(.system(size: 20)).foregroundColor(VoiidColor.primary)
            Text("Voiid AI").font(VoiidFont.headline).foregroundColor(VoiidColor.textPrimary)
            Spacer()
        }
        .padding(.horizontal, VoiidSpacing.md).padding(.vertical, VoiidSpacing.sm)
        .background(.ultraThinMaterial).overlay(Divider(), alignment: .bottom)
    }

    private func aiBubble(_ m: VAIMessage) -> some View {
        HStack {
            if m.isUser { Spacer(minLength: 40) }
            Text(m.text)
                .font(VoiidFont.body).foregroundColor(VoiidColor.textPrimary)
                .padding(.horizontal, VoiidSpacing.md).padding(.vertical, VoiidSpacing.sm)
                .background(m.isUser ? VoiidColor.bubbleSent : VoiidColor.bubbleReceived)
                .clipShape(BubbleShape(isMine: m.isUser))
            if !m.isUser { Spacer(minLength: 40) }
        }
        .transition(.scale(scale: 0.85, anchor: m.isUser ? .bottomTrailing : .bottomLeading).combined(with: .opacity))
    }

    private var inputBar: some View {
        VStack(spacing: 0) {
            Divider()
            HStack(spacing: VoiidSpacing.sm) {
                TextField("Ask Voiid AI…", text: $draft, axis: .vertical)
                    .font(VoiidFont.body).lineLimit(1...4)
                    .padding(.horizontal, VoiidSpacing.md).frame(minHeight: 44)
                    .background(VoiidColor.fieldFill)
                    .clipShape(RoundedRectangle(cornerRadius: VoiidRadius.pill, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: VoiidRadius.pill).stroke(VoiidColor.fieldBorder))
                Button {
                    let t = draft.trimmingCharacters(in: .whitespaces)
                    guard !t.isEmpty else { return }
                    Haptics.tap(); ai.send(t); draft = ""
                } label: {
                    Image(systemName: "arrow.up.circle.fill").font(.system(size: 32)).foregroundColor(VoiidColor.primary)
                }
            }
            .padding(.horizontal, VoiidSpacing.sm).padding(.vertical, VoiidSpacing.sm)
            .padding(.bottom, 80)
            .background(.ultraThinMaterial)
        }
    }
}

File: apps/android/app/src/main/java/com/voiid/app/main/AIChatView.kt
package com.voiid.app.main

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.lazy.rememberLazyListState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowCircleUp
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.dp
import com.voiid.app.model.AIStore
import com.voiid.app.model.VAIMessage
import com.voiid.app.ui.components.LocalVoiidHaptics
import com.voiid.app.ui.theme.VoiidColor
import com.voiid.app.ui.theme.VoiidFont
import com.voiid.app.ui.theme.VoiidRadius

/** In-app AI assistant (Figma Screen-8) — port of `AIChatView.swift`. */
@Composable
fun AIChatView(ai: AIStore) {
    val haptics = LocalVoiidHaptics.current
    var draft by remember { mutableStateOf("") }
    val listState = rememberLazyListState()

    LaunchedEffect(ai.messages.size, ai.thinking) {
        val count = ai.messages.size + if (ai.thinking) 1 else 0
        if (count > 0) listState.animateScrollToItem(count - 1)
    }

    Column(Modifier.fillMaxSize().background(VoiidColor.background)) {
        // Header
        Row(
            Modifier
                .fillMaxWidth()
                .background(VoiidColor.background)
                .statusBarsPadding()
                .padding(horizontal = 16.dp, vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            Icon(Icons.Default.AutoAwesome, null, tint = VoiidColor.primary, modifier = Modifier.size(20.dp))
            Text("Voiid AI", style = VoiidFont.headline, color = VoiidColor.textPrimary)
        }
        Box(Modifier.fillMaxWidth().height(1.dp).background(VoiidColor.divider.copy(alpha = 0.5f)))

        // Messages
        LazyColumn(
            state = listState,
            modifier = Modifier.fillMaxWidth().weight(1f),
            contentPadding = androidx.compose.foundation.layout.PaddingValues(horizontal = 16.dp, vertical = 16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            items(ai.messages, key = { it.id }) { m -> AiBubble(m) }
            if (ai.thinking) item(key = "thinking") { TypingBubble() }
        }

        // Input bar
        Box(Modifier.fillMaxWidth().height(1.dp).background(VoiidColor.divider.copy(alpha = 0.5f)))
        Row(
            Modifier
                .fillMaxWidth()
                .background(VoiidColor.background)
                .padding(horizontal = 8.dp, vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            val fieldShape = RoundedCornerShape(VoiidRadius.pill)
            Box(
                Modifier
                    .weight(1f)
                    .heightIn(min = 44.dp)
                    .clip(fieldShape)
                    .background(VoiidColor.fieldFill)
                    .border(1.dp, VoiidColor.fieldBorder, fieldShape)
                    .padding(horizontal = 16.dp, vertical = 10.dp),
                contentAlignment = Alignment.CenterStart,
            ) {
                if (draft.isEmpty()) Text("Ask Voiid AI…", style = VoiidFont.body, color = VoiidColor.placeholder)
                BasicTextField(
                    value = draft,
                    onValueChange = { draft = it },
                    textStyle = VoiidFont.body.merge(TextStyle(color = VoiidColor.textPrimary)),
                    cursorBrush = SolidColor(VoiidColor.primary),
                    maxLines = 4,
                    modifier = Modifier.fillMaxWidth(),
                )
            }
            Icon(
                Icons.Default.ArrowCircleUp, "Send", tint = VoiidColor.primary,
                modifier = Modifier.size(32.dp).clickable {
                    val t = draft.trim()
                    if (t.isNotEmpty()) { haptics.tap(); ai.send(t); draft = "" }
                },
            )
        }
    }
}

@Composable
private fun AiBubble(m: VAIMessage) {
    Row(Modifier.fillMaxWidth()) {
        if (m.isUser) Spacer(Modifier.weight(1f))
        Text(
            m.text,
            style = VoiidFont.body,
            color = VoiidColor.textPrimary,
            modifier = Modifier
                .widthIn(max = 300.dp)
                .clip(bubbleShape(m.isUser))
                .background(if (m.isUser) VoiidColor.bubbleSent else VoiidColor.bubbleReceived)
                .padding(horizontal = 16.dp, vertical = 8.dp),
        )
        if (!m.isUser) Spacer(Modifier.weight(1f))
    }
}
